manticore/utils/event.py

```diff
@@ -139,22 +139,19 @@ def _publish_impl(self, _name, *args, **kwargs):
                 sink._publish_impl(_name, *args, **kwargs)
 
     def subscribe(self, name, method):
-        if not inspect.ismethod(method):
-            raise TypeError
+        assert inspect.ismethod(method), f'{method.__class__.__name__} is not a method'
```

Comment:
i'm not sure if asserts are more appropriate for this situation? these are not invariants, it's more argument checking, so i think raising TypeErrors might be better (although the error messages could be improved)

Reply:
Do our users call `subscribe`? It seems to me that subscribing to an event is an internal thing and the events are exposed to users only through plugins/detectors.

If it is internal - then having an assertion here should be good enough ( ? ).

Reply:
you're right, generally users should opt for the Plugin/Detector interface rather than using the raw subscribe one. however, i'm not sure whether this is an external or internal interface matters too much; imo, it's more about whether the condition being checked is an invariant or for input error checking. in this case, it seems more for error checking, and `TypeError` actually seems like a more semantically correct and descriptive exception to raise, since this is actually a type check

Reply:
Honestly I think that in the end ("in a production build") such checks are not needed (the code itself should never call this with bad type) but it is good to have it as a defence against ourselves when making changes. This way, having it as assertion is better, at least in theory (I wonder if there is a way to ship code without assertions).

Reply:
yeah, I see your point. i think for this particular case it doesn't matter too much, i don't think `subscribe` is in any hot code paths. at the end of the day i think i do slightly prefer the TypeError since it is more specific, and from what i can see, having it shouldn't affect performance in a noticeable way. but it's only a slight preference :)